docs/new-resource.md

```diff
@@ -1,149 +1,217 @@
+
+Bear in mind it will be called by the Enumerator to retrieve the list of instances.
+
+Enumerator constructor could use these arguments:
+- an instance of `Repository` that you will use to retrieve information about the resource
+- the global resource factory that should always be used to create a new `resource.Resource`
+
+Enumerator then needs to implement:
+- `SupportedType() resource.ResourceType` that will return the constant you defined in the type file
+- `Enumerate() ([]*resource.Resource, error)` that will return the list of resources
 
 ```go
-func Deserializers() []remote.CTYDeserializer {
-	return []remote.CTYDeserializer{
-        aws.NewS3BucketDeserializer(),
-		...
+type EC2InstanceEnumerator struct {
+	repository repository.EC2Repository
+	factory    resource.ResourceFactory
+}
+
+func NewEC2InstanceEnumerator(repo repository.EC2Repository, factory resource.ResourceFactory) *EC2InstanceEnumerator {
+	return &EC2InstanceEnumerator{
+		repository: repo,
+		factory:    factory,
 	}
 }
-```
 
-Then in the cloud provider's init file (e.g. in `remote/aws/init.go`), add your new implementation for `resource.Supplier`:
+func (e *EC2InstanceEnumerator) SupportedType() resource.ResourceType {
+	return aws.AwsInstanceResourceType
+}
 
-```go
-func Init() error {
-	provider, err := NewTerraFormProvider()
+func (e *EC2InstanceEnumerator) Enumerate() ([]*resource.Resource, error) {
+	instances, err := e.repository.ListAllInstances()
 	if err != nil {
-		return err
+		return nil, remoteerror.NewResourceListingError(err, string(e.SupportedType()))
+	}
+
+	results := make([]*resource.Resource, len(instances))
+
+	for _, instance := range instances {
+		results = append(
+			results,
+			e.factory.CreateAbstractResource(
+				string(e.SupportedType()),
+				*instance.InstanceId,
+				map[string]interface{}{},
+			),
+		)
 	}
 
-	terraform.AddProvider(terraform.AWS, provider)
-	resource.AddSupplier(NewS3BucketSupplier(provider.Runner().SubRunner(), s3.New(provider.session)))
-	...
+	return results, err
 }
 ```
 
-Don't forget to add unit tests after adding a new resource.
-You can also add acceptance tests if you think it makes sense.
+As you can see, listing error are treated in a particular way. Instead of failing and stopping the scan they will be handled, and an alert will be created.
+So please don't forget to wrap these errors inside a `NewResourceListingError`.
+For some provider error handling is not that coherent, so you might need to check in `pkg/remote/resource_enumeration_error_handler.go` and add a new case for your error.
+You should test enumerator behavior when you do not have permission to enumerate resource, in the snippet above, `ListAllInstances` may return an `AccessDenied` error or so that should be handled.
+
+Once the enumerator is written you have to add it to the remote init located in `pkg/remote/<providername>/init.go` :
```

Comment:
`to the remote initialization located in pkg/remote/<providername>/init.go:`
Newline